### DISCUSSION TITLE
Something is broken after updating to latest with changes for the colorscheme

### DISCUSSION THREAD

**Discussion by appelgriebsch** [LazyVim/LazyVim#31] [Q&A] [answered]
I followed the example to update my configuration to use the new mechanism for setting the color scheme.... like so:

```lua
  {
    "projekt0n/github-nvim-theme",
  },

  -- Configure LazyVim to load github-dark color scheme
  {
    "LazyVim/LazyVim",
    opts = {
      colorscheme = "github_dark",
    },
  },
```

Find two issues with it:

1. when starting neovim I got the following message 

```
`config.settings` is deprecated.

Please configure LazyVim as a plugin:

{
  "LazyVim/LazyVim",
  opts = {
    -- your config comes here
  },
}

Press ENTER or type command to continue
```

2. how would I configure special options for the color scheme (such as those mentioned here: https://github.com/projekt0n/github-nvim-theme#normal)? Do I have to use the "vim script" way? As providing an opts = {} to the GitHub-theme plugin section doesn't seem to work.... so this is broken:

```
  {
    "projekt0n/github-nvim-theme",
    opts = {
      comment_style = "NONE",
      keyword_style = "NONE",
      function_style = "NONE",
      variable_style = "NONE"
      -- other config
    },
    config = true
  },
```

**Comment by appelgriebsch** [comment, marked as answer]
Ok, got the hint for 2nd issue.... colorscheme could also be a lua function....

**Reply by appelgriebsch** [reply]
Also found the source for the 1st message.... in my lualine setup I have this...

```
      local icons = require("lazyvim.config.settings").icons
```

to get access to the pre-defined icons for diagnostics...

**Reply by appelgriebsch** [reply]
ok, changing it to 

```
local icons = require("lazyvim.config").icons
```

like in the original ui.lua solved it.

**Comment by folke** [comment]
Sorry about breaking your config. Once `1.0.0` gets released, that won't happen (frequently) again.

The deprecation of  `settings` was a warning. That actually still works for now.

For your colorscheme, what you did first, was actually correct, however that colorscheme is a special case.

When doing `config=true` or `opts=...`, lazy tries to find the correct module that has the `setup()` method. This works fine for most plugins, but fails for some, like yours. Reason is that `github-nvim-theme`, is too different from the lua module `github-theme`.

In those cases, you can still do the correct setup yourself. So setting colorscheme for LazyVim to `github-dark` will work just fine, but you need to change the spec for the colorscheme like so:

```lua
  {
    "projekt0n/github-nvim-theme",
    config = function()
      require("github-theme").setup({
        comment_style = "NONE",
        keyword_style = "NONE",
        function_style = "NONE",
        variable_style = "NONE",
      })
    end,
  },
```

**Reply by appelgriebsch** [reply]
No worries. Was a bit of a surprise in the beginning, but at the end not much changes to my custom setup….(once I figured the details).

on the plus side one can customize the icons now as well. So at the end a win-win for everyone ;)
